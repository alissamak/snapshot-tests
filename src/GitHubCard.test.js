import { render, screen, waitFor } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import GitHubCard from './components/GitHubCard';

test('renders a snapshot', () => {
    const tree = renderer.create(<GitHubCard/>).toJSON()
    expect(tree).toMatchSnapshot()
    
  });

  beforeEach(()=> {
    //sorts eveyrthing back to initial state before each test
    fetch.resetMocks();
})

test('fetch gitHub profile name from GitHub REST API using jest fetch mock', async () => {
    fetch.mockResponseOnce(JSON.stringify({name: 'Alissa M'}))
    render(<GitHubCard/>)
    const gitHubName = await waitFor(() => screen.getByRole('card-title'))
    expect(gitHubName).toHaveTextContent('Alissa M')
})