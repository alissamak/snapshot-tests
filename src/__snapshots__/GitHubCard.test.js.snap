// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App pt-5"
>
  <div
    className="col"
  >
    <div
      className="row"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <div
        className="card"
      >
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Github profile image"
            className="card-img-top"
            height="200px"
            src=""
            width="200px"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            />
            <p
              className="card-text"
            >
              <p />
              <a
                href=""
              >
                <button>
                  Link to GitHub Profile
                </button>
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
